### PR TITLE
Adjust pagination to use X-Next-Page

### PR DESCRIPTION
**Pull request by Dagg-h**
### Issue number

DATA 1593

### Expected behaviour

Retrieve all commits

### Actual behaviour

Only retrieve 50 commits.

### Description of fix

Adjust pagination to use different response header.

### CODE REVIEW

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 1]
```diff
@@ -58,8 +58,8 @@ def format_url(
         """
         updated_since = f"&updated_after={since}" if since else ""
         sort_string = f"&sort=asc" if sort else ""
```
Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py]: There's no variable, so no need for an f-string

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 2]
```diff
@@ -58,8 +58,8 @@ def format_url(
         """
         updated_since = f"&updated_after={since}" if since else ""
```
Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py:62]: suggest to add a an example format on line 54/55 & 85

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 3]
```diff
@@ -118,12 +116,10 @@ def paginate_through_resource(
...
                 all_resources.append(list(final_json.values()))
-
             try:
-                total_pages = int(response.headers["X-Total-Pages"])
-                current_page = int(response.headers["X-Page"])
-            except:
-                current_page = current_page + 1
+                next_page = response.headers['X-Next-Page']
+            except Exception as e:
+                print(e)
```
Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py]: with `.get()` you do not need a try / except

File: wherescape/connectors/gitlab/gitlab_wrapper.py (hunk 2 above)
Comment (BFJonk) [reply wherescape/connectors/gitlab/gitlab_wrapper.py:62]: Apologies for not being clear: The example line inserted does not contain an ISO-formatted (I think one makes things easier to use, if you have an example format of a timestamp).

**Comment by BFJonk** [comment]
It's nice to have a commit per review comment (Never used it that way).

File: wherescape/connectors/gitlab/gitlab_wrapper.py (hunk 2 above)
Comment (Dagg-h) [reply wherescape/connectors/gitlab/gitlab_wrapper.py:62]: I have put updated_after in the example return, but also put examples at the parameter itself!

**Pull request merged**